python/tskit/drawing.py

```diff
@@ -577,38 +606,89 @@ def assign_x_coordinates(self, tree, x_start, width):
                         node_x_coord_map[u] = a + (b - a) / 2
         return node_x_coord_map
 
-    def info_classes(self, focal_node):
+    def add_node(self, curr_svg_group, focal, dx, dy):
         """
-        For a focal node id, return a set of classes that encode this useful information:
-        "nA":           where A == focal node id
-        "pB" or "root": where B == parent id (or "root" if the focal node is a root)
-        "sample":       a class present if the focal node is a sample
-        "leaf":         a class present if the focal node is a leaf
-        "mC":           where C == mutation id of all mutations above this focal node
-        "sD":           where D == site id of the sites associated with all mutations
-                            above this focal node
+        Return a list of SvgGroupInfo objects to add to the stack
         """
-        # Add a new group for each node, and give it classes for css targetting
-        classes = set()
-        classes.add(f"node n{focal_node}")
-        v = self.tree.parent(focal_node)
+        ret = []
+        dwg = self.drawing
+        grp = curr_svg_group
+        v = self.tree.parent(focal)
+        classes = [f"n{focal}"]
+        offset_x = dx
+        offset_y = dy
         if v == NULL:
-            classes.add(f"root")
+            classes.append(f"root")
+            if self.root_branch:
```

Comment:
I don't think this if statement does anything - you've computed the ``self.root_branch_length`` above so there's no need for the local variable.